scripts/globals/weaponskills.lua

```diff
@@ -869,11 +869,14 @@ end
 -- handles displaying the appropriate action/message, delivering the damage to the mob, and any enmity from it
 function takeWeaponskillDamage(defender, attacker, wsParams, primaryMsg, attack, wsResults, action)
     local finaldmg = wsResults.finalDmg
+    local targetTPMult = wsParams.targetTPMult or 1
+    finaldmg = defender:takeWeaponskillDamage(attacker, finaldmg, attack.type, attack.damageType, attack.slot, primaryMsg, wsResults.tpHitsLanded, (wsResults.extraHitsLanded * 10) + wsResults.bonusTP, targetTPMult)
 
     if wsResults.tpHitsLanded + wsResults.extraHitsLanded > 0 then
         if finaldmg >= 0 then
             if primaryMsg then
                 action:messageID(defender:getID(), xi.msg.basic.DAMAGE)
+                action:param(defender:getID(), math.abs(finaldmg))
```

Comment:
possibly also needed in the block at line 881 ? (I am legit not sure)

Reply:
I'll test thoroughly to be 100% sure, but I think that both of those messages use the one param.

Reply:
Testing with circle blade, DAMAGE_SECONDARY does also need the finaldmg param. Thanks!